### ISSUE TITLE
[feature] security scanning

### ISSUE THREAD

**Issue by vsoch** [singularityhub/sregistry#128]
We should have security scanning that is akin to a plugin, looking like this: https://hub.docker.com/r/library/busybox/tags/

**Comment by vsoch** [comment]
For this hacktoberfest issue, contribute a plugin that will add a claire security scanner, likely as another docker-compose image that can be optionally added by way of activating the plugin and switching to a modified docker-compose file.

**Comment by vladdoster** [comment]
HMS team is working on this use-case. It seems this isn't resolved, correct?

**Comment by vsoch** [comment]
Correct! The idea of a security scanner would mean that the containers that are pushed are always scanned, and then the report is shown. This would need to be scaled so it's a non trivial issue, at least for any registry that is larger than a few users. There are likely other scanners / solutions now, but back when I wrote the issue what I had tried was modifying Clair to be used for Singularity. You can see an example at https://github.com/singularityhub/stools.

**Comment by acullenhms** [comment]
We've developed a pipeline in house to execute stools scans and our own custom scan sequence. We're now looking at porting that over to an sregistry plugin.

When you say

>This would need to be scaled

I read this as: Infrastructure demands for lots of concurrent scans would need to be accounted for. Including how we deploy many concurrent workers to perform the scans without interfering with each other.

Am I interpreting this correctly?

**Comment by vsoch** [comment]
Yes that's correct. For a registry for one person, a service running alongside the registry (that only scans a few containers a day) might be appropriate. For a registry with much higher demand, it likely wouldn't be because of filesystem space and memory needed to run concurrent scans.
